lib.php

```diff
@@ -625,23 +625,20 @@ function local_boostnavigation_extend_navigation(global_navigation $navigation)
 
     // Add calendar link after grades icon
     $coursehomenode = $PAGE->navigation->find($COURSE->id, navigation_node::TYPE_COURSE);
-    if(empty($coursehomenode)) {
-
-        $calendarnode = navigation_node::create(get_string('calendar', 'calendar'),
-            new moodle_url('/calendar/view.php', array('course' => $COURSE->id, 'view' => 'month')),
-            global_navigation::TYPE_CUSTOM,
-            null,
-            'calendar',
-            new pix_icon('i/calendar', '')
-        );
-        $coursehomenode->add_node($calendarnode, 'localboostnavigationcoursesections');
-        $calendarurlcheck = strpos($PAGE->url->out(), 'calendar/view.php?course=' . $COURSE->id);
-        if ($calendarurlcheck) {
-            $calendarnode->isactive = true;
-            // Marking the additional course node at the bottom as inactive.
-            $coursehomenode = $PAGE->navigation->find($COURSE->id, navigation_node::TYPE_COURSE);
-            $coursehomenode->isactive = false;
-        }
+    $calendarnode = navigation_node::create(get_string('calendar', 'calendar'),
+        new moodle_url('/calendar/view.php', array('course' => $COURSE->id, 'view' => 'month')),
+        global_navigation::TYPE_CUSTOM,
+        null,
+        'calendar',
+        new pix_icon('i/calendar', '')
+    );
+    $coursehomenode->add_node($calendarnode, 'localboostnavigationcoursesections');
```

Comment:
Do you think a check is needed to see if a calendar-node already exists?

Reply:
I tried, it's not the issue the current fix removes it from the courses page and LT wants it back now. the original calendar node is hidden via CSS, so would need to remove the previous calendar node on course pages only.

Reply:
As long as we ensure that there is not a chance for 2 calendar nodes then I am happy.